src/Pos/Wallet/Web/Methods/Payment.hs

```diff
@@ -0,0 +1,159 @@
+import           Pos.Wallet.Web.Error           (WalletError (..))
+import           Pos.Wallet.Web.Methods.History (addHistoryTx)
+import qualified Pos.Wallet.Web.Methods.Logic   as L
+import           Pos.Wallet.Web.Mode            (MonadWalletWebMode, WalletWebMode)
+import           Pos.Wallet.Web.State           (AddressLookupMode (Existing))
+import           Pos.Wallet.Web.Util            (coinDistrToOutputs, decodeCTypeOrFail,
+                                                 getAccountAddrsOrThrow,
+                                                 getWalletAccountIds, rewrapTxError)
+
+
+newPayment
+    :: MonadWalletWebMode m
+    => SendActions m
+    -> PassPhrase
+    -> AccountId
+    -> CId Addr
+    -> Coin
+    -> m CTx
+newPayment sa passphrase srcAccount dstAccount coin =
+    sendMoney
+        sa
+        passphrase
+        (AccountMoneySource srcAccount)
+        (one (dstAccount, coin))
+
+getTxFee
+     :: MonadWalletWebMode m
+     => AccountId
+     -> CId Addr
+     -> Coin
+     -> m CCoin
+getTxFee srcAccount dstAccount coin = do
+    utxo <- getMoneySourceUtxo (AccountMoneySource srcAccount)
+    outputs <- coinDistrToOutputs $ one (dstAccount, coin)
+    TxFee fee <- eitherToThrow =<< runExceptT (computeTxFee utxo outputs)
+    pure $ mkCCoin fee
+
+data MoneySource
+    = WalletMoneySource (CId Wal)
+    | AccountMoneySource AccountId
+    | AddressMoneySource CWAddressMeta
+    deriving (Show, Eq)
+
+getMoneySourceAddresses :: MonadWalletWebMode m => MoneySource -> m [CWAddressMeta]
+getMoneySourceAddresses (AddressMoneySource addrId) = return $ one addrId
+getMoneySourceAddresses (AccountMoneySource accId) =
+    getAccountAddrsOrThrow Existing accId
+getMoneySourceAddresses (WalletMoneySource wid) =
+    getWalletAccountIds wid >>=
+    concatMapM (getMoneySourceAddresses . AccountMoneySource)
+
+getSomeMoneySourceAccount :: MonadWalletWebMode m => MoneySource -> m AccountId
+getSomeMoneySourceAccount (AddressMoneySource addrId) =
+    return $ addrMetaToAccount addrId
+getSomeMoneySourceAccount (AccountMoneySource accId) = return accId
+getSomeMoneySourceAccount (WalletMoneySource wid) = do
+    wAddr <- (head <$> getWalletAccountIds wid) >>= maybeThrow noWallets
+    getSomeMoneySourceAccount (AccountMoneySource wAddr)
+  where
+    noWallets = InternalError "Wallet has no accounts"
+
+getMoneySourceWallet :: MoneySource -> CId Wal
+getMoneySourceWallet (AddressMoneySource addrId) = cwamWId addrId
+getMoneySourceWallet (AccountMoneySource accId)  = aiWId accId
+getMoneySourceWallet (WalletMoneySource wid)     = wid
+
+getMoneySourceUtxo :: MonadWalletWebMode m => MoneySource -> m Utxo
+getMoneySourceUtxo =
+    getMoneySourceAddresses >=>
+    mapM (decodeCTypeOrFail . cwamId) >=>
+    getOwnUtxos
+
+-- [CSM-407] It should be moved to `Pos.Wallet.Web.Mode`, but
+-- to make it possible all this mess should be neatly separated
+-- to modules and refactored
```

Comment:
Heey, actually _now_ you can move it to `Pos.Wallet.Web.Mode`, if I'm not mistaken - because you moved `newAddress` to a separate `Logic` module.